test/e2e/framework/autoscaling/autoscaling_utils.go

```diff
@@ -455,6 +507,10 @@ func runServiceAndWorkloadForResourceConsumer(c clientset.Interface, ns, name st
 		Annotations: podAnnotations,
 	}
 
+	if len(additionalContainers) > 0 {
+		rcConfig.AdditionalContainers = additionalContainers
+	}
```

Comment:
I don't think you need to check the length of `additionalContainers` here. You should be able to set it directly when creating the `rcConfig` even if it is empty without encountering trouble.

Reply:
Removed length check